### PR TITLE
Iterable bug issue

### PR DESCRIPTION
**Pull request by rohithkaza**
## Description of the change

The code works well for the “String” data type. But generates an incorrect response for other data types.
Fixing the error by adding new conditional cases for other data types.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by saikumarrs — DISMISSED** [review]

File: v0/destinations/iterable/transform.js [hunk 1]
```diff
@@ -147,7 +147,7 @@ function constructPayloadItem(message, category, destination) {
             el,
             mappingConfig[ConfigCategory.PRODUCT.name]
           );
-          if (element.categories) {
+          if (element.categories && typeof element.categories === "stirng") {
```
Comment (akashrpo) [inline v0/destinations/iterable/transform.js:150]:
```suggestion
          if (element.categories && typeof element.categories === "string") {
```
can you please correct the string name and same applies for other changes below

File: v0/destinations/iterable/transform.js [hunk 2]
```diff
@@ -160,7 +160,7 @@ function constructPayloadItem(message, category, destination) {
           message.properties,
           mappingConfig[ConfigCategory.PRODUCT.name]
         );
-        if (element.categories) {
+        if (element.categories && typeof element.categories === "stirng") {
```
Comment (akashrpo) [inline v0/destinations/iterable/transform.js:163]: here as well

**Comment by akashrpo** [comment]
Please revert the `package-lock.json` file. We should not delete this and the diff is −9,843 above.

File: v0/destinations/iterable/transform.js (hunk 2 above)
Comment (rohithkaza) [reply v0/destinations/iterable/transform.js]: Done. Please review

**Review by akashrpo — DISMISSED** [review]
LGTM

**Review by utsabc — APPROVED** [review]

File: __tests__/data/iterable_output.json [hunk 3]
```diff
@@ -503,7 +503,7 @@
             "id": "507f1f77bcf86cd799439011",
             "sku": "45790-32",
             "name": "Monopoly: 3rd Edition",
-            "categories": ["cars"],
+            "categories": "cars",
```
[the remark this answers is not in the document]
Comment (utsabc) [reply __tests__/data/iterable_output.json]: Only arrays are present in test cases please add a case where `split` fxn would be in effect

[the remark this answers is not in the document]
Comment (rohithkaza) [reply __tests__/data/iterable_output.json]: I changed the test case on the 7th commit- https://github.com/rudderlabs/rudder-transformer/pull/1024/commits/f4b0ebacb1a71de46b6a2e067e7f110c4274a590
Please verify

[the remark this answers is not in the document]
Comment (rohithkaza) [reply __tests__/data/iterable_output.json]: These are the new test cases that are added:
"category": ["bikes", "cars", "motors"]---------- o/p: "categories": ["bikes", "cars", "motors"]
"category": "shirts,pants,trousers"------------- o/p: "categories": ["shirts", "pants", "trousers"]

[the remark this answers is not in the document]
Comment (utsabc) [reply __tests__/data/iterable_output.json]: Got it, resolving the conversation

**Review by utsabc — DISMISSED** [review]

File: v0/destinations/iterable/transform.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ const {
   isAppleFamily
 } = require("../../util");
 const logger = require("../../../logger");
+const { string } = require("is");
```
Comment (akashrpo) [inline v0/destinations/iterable/transform.js:16]: Do we require this import?

Comment (akashrpo) [reply v0/destinations/iterable/transform.js:16]: Let's remove this

**Review by akashrpo — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**